src/plugins/grpc/server.js

```diff
@@ -0,0 +1,137 @@
+'use strict'
+
+const { FORMAT_HTTP_HEADERS, Tags } = require('opentracing')
+
+let patchedHandlers = []
+
+function createWrapHandler (grpc, tracer, config, handler) {
+  return function wrapHandler (func) {
+    return function funcHandler (emitter) {
+      const { metadata, request } = emitter
+      const { type } = this
+      const isStream = ['client_stream', 'server_stream'].includes(type)
+      const childOf = tracer.extract(FORMAT_HTTP_HEADERS, metadata.getMap())
+      const span = tracer.startSpan('grpc.server', {
+        childOf,
+        tags: {
+          [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_SERVER,
+          'grpc.method_name': handler,
+          'grpc.method_type': type,
+          'resource.name': handler.split('/').slice(-1),
+          'service.name': config.service || `${tracer._service}-grpc`
+        }
+      })
+
+      tracer.scopeManager().activate(span)
+
+      if (request) {
+        if (metadata) {
+          for (const [key, value] of Object.entries(metadata.getMap())) {
+            span.setTag(`grpc.metadata.${key}`, value)
```

Comment:
Can metadata contain sensitive data?

Reply:
Potentially, yes. It's user input as far this lib is concerned. I see what you're saying.

I'll add another callback to the config option.